rcl_logging_spdlog/test/test_logging_interface.cpp

```diff
@@ -162,3 +163,16 @@ TEST_F(LoggingTest, full_cycle)
     expected_log.str(),
     actual_log.str()) << "Unexpected log contents in " << log_file_path;
 }
+
+TEST_F(LoggingTest, init_fini_maybe_fail_test)
+{
+  RCUTILS_FAULT_INJECTION_TEST(
+  {
+    if (RCL_LOGGING_RET_OK == rcl_logging_external_initialize(nullptr, allocator)) {
+      EXPECT_EQ(RCL_LOGGING_RET_OK, rcl_logging_external_shutdown());
```

Comment:
@brawner what if `rcl_logging_external_shutdown()` fails because of fault injection?

Reply:
`rcl_logging_external_shutdown()` is currently very simple and doesn't call any external functions. The `EXPECT_EQ` will help notify people this test needs to be fixed should that function be modified.

https://github.com/ros2/rcl_logging/blob/master/rcl_logging_spdlog/src/rcl_logging_spdlog.cpp#L150-L154